c7n/resolver.py

```diff
@@ -40,12 +44,21 @@ def resolve(self, uri):
         else:
             # TODO: in the case of file: content and untrusted
             # third parties, uri would need sanitization
-            fh = urlopen(uri)
-            contents = fh.read().decode('utf-8')
-            fh.close()
+            req = Request(uri, headers={"Accept-Encoding": "gzip"})
+            with closing(urlopen(req)) as response:
+                contents = self.handle_response_encoding(response)
+
         self.cache.save(("uri-resolver", uri), contents)
         return contents
 
+    def handle_response_encoding(self, response):
+        if response.info().get('Content-Encoding') != 'gzip':
```

Comment:
the detection here needs to be a bit more robust to be useful for other scenarios. is the only scenario your trying to address gzip encoding on http(s) url?  ie passing in a file url sets content-type not encoding, and zip would be application/zip on content type.

Reply:
For our immediate use case yes. I can give file handling a try. If you have certain test cases in mind please post/lmk.

Reply:
let's just keep as is for now, the other scenarios are where the file is actually compressed (ie. .gz), not compressed via encoding, and handling that across urls and object storage setups.